parsec/utils.py

```diff
@@ -225,13 +227,12 @@ def collapse_exception_group(exception_group: BaseExceptionGroup[E]) -> Exceptio
         # Craft the specific class, inheriting from BaseExceptionGroup
         cls: Any = type(name, (type(pick), BaseExceptionGroup), attrs)
         # Instantiate the instance
-        result = cls()
+        result = cls(exception_group.message, exception_group.exceptions)
```

Comment:
This is probably more robust:
```suggestion
        result = BaseExceptionGroup.__new__(cls, exception_group.message, exception_group.exceptions)
```